types/react-native-safe-area/index.d.ts

```diff
@@ -0,0 +1,48 @@
+// Type definitions for react-native-safe-area 0.5
+// Project: https://github.com/miyabi/react-native-safe-area#readme
+// Definitions by: Pavlos Vinieratos <https://github.com/pvinis>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
+
+/// <reference types="react" />
+
+import { EmitterSubscription } from 'react-native';
+
+// from `TypeDefinition.js`
+export interface SafeAreaInsets { top: number; left: number; bottom: number; right: number; }
+
+// from `SafeArea.[ios|android].js`
+declare namespace SafeArea {
+    function getSafeAreaInsetsForRootView(): Promise<{ safeAreaInsets: SafeAreaInsets }>;
+    function addEventListener(eventType: string, listener: (...args: any[]) => any, context?: any): EmitterSubscription;
+    function removeEventListener(eventType: string, listener: (...args: any[]) => any): void;
+}
+export default SafeArea;
+
+// from `withSafeArea.js`
+export type Direction =
+    | 'top'
+    | 'bottom'
+    | 'left'
+    | 'right'
+    | 'topLeft' // DEPRECATED
+    | 'topAndLeft'
+    | 'topRight' // DEPRECATED
+    | 'topAndRight'
+    | 'bottomLeft' // DEPRECATED
+    | 'bottomAndLeft'
+    | 'bottomRight' // DEPRECATED
+    | 'bottomAndRight'
+    | 'vertical'
+    | 'horizontal'
+    | 'verticalAndLeft'
+    | 'verticalAndRight'
+    | 'horizontalAndTop'
+    | 'horizontalAndBottom'
+    | 'all';
+
+export function withSafeArea(
+    WrappedComponent: React.ComponentType<any>,
+    applyTo: 'margin' | 'padding' | 'absolutePosition' | 'contentInset' = 'margin',
```

Comment:
The `= 'margin'` part of this is not allowed in a .d.ts file.
You should instead use `applyTo?: 'margin' | 'padding' | 'absolutePosition' | 'contentInset'`
You should be able to use JSDoc comments to document the default value that is provided if you'd like.

Reply:
done